File: Conit.DAL/Entities/Product.cs
namespace Conit.DAL.Entities
{
    public class Product
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public int CompanyId { get; set; }

        public Company Company { get; set; }

        public string Category { get; set; }

        public string PictureId { get; set; }

        public bool IsDeleted { get; set; }
    }
}

File: Mobile/flutter_app/flutter_app/lib/product.dart
class Product {
  final int id;
  final String name;
  final String description;
  final String category;
  final String pictureId;

  Product._({
    this.id,
    this.name,
    this.description,
    this.category,
    this.pictureId
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return new Product._(
        id: json['id'],
        name: json['name'],
        description: json['description'],
        category: json['category'],
        pictureId: json['pictureId']
    );
  }
}